src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -8662,13 +8626,36 @@ namespace WaterThermalTanks {
         Array1D<Real64> Par(5);     // Parameters passed to RegulaFalsi
         static Real64 MinTemp(0.0); // used for error messages, C
         std::string IterNum;        // Max number of iterations for warning message
+        Real64 TimeElapsed = HourOfDay + TimeStep * TimeStepZone + SysTimeElapsed;
 
         // FLOW:
         DesuperheaterNum = WaterThermalTank(WaterThermalTankNum).DesuperheaterNum;
         AvailSchedule = GetCurrentScheduleValue(WaterHeaterDesuperheater(DesuperheaterNum).AvailSchedPtr);
         WaterInletNode = WaterHeaterDesuperheater(DesuperheaterNum).WaterInletNode;
         WaterOutletNode = WaterHeaterDesuperheater(DesuperheaterNum).WaterOutletNode;
 
+        if (DesuperheaterTimeElapsed != TimeElapsed && WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSource == COIL_AIR_WATER_HEATPUMP_EQ) {
+            DataHeatBalance::HeatReclaimSimple_WAHPCoil(WaterHeaterDesuperheater(DesuperheaterNum).ReclaimHeatingSourceIndexNum).DesuperheaterReclaimedHeat = 0;
```

Comment:
I tried to find the best place to initialize this variable, I think the most ideal place would be somewhere at the top level before hitting any specific functions (because I didn't know which part would be hit first and if it could be a "user" of reclaimed heat)